### ISSUE TITLE
(check-unsat-cores) Segmentation fault (sat_proof_implementation.h:795)

### ISSUE THREAD

**Issue by muchang** [cvc5/cvc5#5908]
```
[584] % cvc4 -q -i --check-unsat-cores small.smt2
sat
sat
CVC4 suffered a segfault.
Offending address is 0x401f2806c
Aborted
[585] % 
[585] % cat small.smt2
(declare-fun a () Int)
(declare-fun b () Int)
(declare-fun c () Int)
(declare-fun d () Int)
(declare-fun e () Int)
(declare-fun f () Int)
(assert (<= 0 a))
(assert (= 0 c))
(assert (not (= a b)))
(assert (= b c))
(assert (not (= d f)))
(check-sat)
(push)
(check-sat)
(pop)
(assert (> e a))
(assert (= e 0))
(check-sat)
[586] %
```
In debug, this triggers the following assertion:
```
Fatal failure within CVC4::ClauseId CVC4::TSatProof<Solver>::resolveUnit(typename Solver::TLit) [with Solver = CVC4::Minisat::Solver; CVC4::ClauseId = unsigned int; typename Solver::TLit = CVC4::Minisat::Lit] at src/./proof/sat_proof_implementation.h:795
Check failure

 reason_ref != Solver::TCRef_Undef

Aborted
```
Commit: 0ff53d7

**Comment by rainoftime** [comment]
Maybe related 
https://github.com/CVC4/CVC4/issues/3651
https://github.com/CVC4/CVC4/issues/4971#issuecomment-703360464

**Comment by HanielB** [comment]
Closed via #6574
